Add internal documentation for thread sampling (#282)
signalfx/signalfx-dotnet-tracing: docs/internal-config.md

## Unpublished settings
| `SIGNALFX_METRICS_ENDPOINT_URL` | The URL to where metric exporters send metrics. | `http://localhost:9943/v2/datapoint` |
| `SIGNALFX_RUNTIME_METRICS_ENABLED` | Enable to activate internal runtime metrics sent to SignalFx. | `false` |

> | `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The Tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |

Reviewer: ```suggestion
| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
```

> | `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |

Reviewer 2: ```suggestion
| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. It is used to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
```

I find the term `tracer` confusing. Especially that here we are talking about metrics. Not traces.

> | `SIGNALFX_THREAD_SAMPLING_ENABLED` | Enable to activate thread sampling sent to SignalFx. | `false` |

Reviewer: ```suggestion
| `SIGNALFX_THREAD_SAMPLING_ENABLED` | Enable to activate thread sampling. | `false` |
```

> | `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period. The tracer uses it to stop threads for a short amount of time and fetch all stack traces. This value cannot be lower that `1000` milliseconds. | `1000` |

Reviewer 2: ```suggestion
| `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period. It definess how often the threads are stopped in order to fetch all stack traces. This value cannot be lower than `1000` milliseconds. | `1000` |
```

I find the term `tracer` confusing. Especially that here we are talking about profiling. Not tracing.

> | `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period to be used. The Tracer uses it to stop threads for shorto amount of time and fetch all stack traces. This value cannot be lower that `1000` ms. | `1000` |

Reviewer: ```suggestion
| `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period. The tracer uses it to stop threads for a short amount of time and fetch all stack traces. This value cannot be lower that `1000` milliseconds. | `1000` |
```
